### PR TITLE
Show message when App goes completely EOL

### PR DESCRIPTION
**Pull request by JakobDev**
We alrdy support Apps changing their ID since #644, but what we currently not support, is when Apps goes completely EOL without a rebase. In this case now a little message is shown, that the App is no longer available (technically you could still install it from console, but I don't think we should say that here) on Flathub together with a Message provided by the App.  So User, we found a Link to a EOL App are know directly told, that this App no longer exists and why instead of confusing them with a 404.

grafik

And yes, I'm a horrible frontend dev.

### CODE REVIEW

**Comment by razzeee** [comment]
Will/should those apps still show up in the search?

**Comment by JakobDev** [comment]
This Apps should of course not show up in the Flathub search. For the Search engines we may need to add `<meta name="robots" content="noindex,nofollow" />` to the header.

**Comment by razzeee** [comment]
https://github.com/garmeeh/next-seo#no-index

**Comment by JakobDev** [comment]
I have now set `noindex`.

I have noticed, that some messages contain a Link, so it might be a good idea to use [linkify-react](https://linkify.js.org/docs/linkify-react.html). It could be used in other parts (e.g. App Descriptions) as well.

**Comment by JakobDev** [comment]
Ping @razzeee

**Comment by razzeee** [comment]
to quote @barthalion 

> I would need to check what are typical messages for apps with eol but without eol-rebase

> just not really sure if there's any point in showing EOL message itself
> "This application is no longer available on Flathub." would be good enough
> let me see if I can dump that from the summary somehow
> this falls into post-launch category, I don't think I want to spend time on thinking about it right now
> the message should make it clear it's a message provided by developer

**Comment by razzeee** [comment]
if your comfortable to join our matrix room, it might be easier to sort this out

https://matrix.to/#/#flathub:matrix.org

**Comment by JakobDev** [comment]
> I would need to check what are typical messages for apps with eol but without eol-rebase

Here's the full list:
```json
{
    "com.beakerbrowser.Beaker":"This application is no longer maintained. The main developer is working on another p2p project. See https://github.com/beakerbrowser/beaker/discussions/1944#discussioncomment-1308779.",
    "com.github.bilelmoussaoui.Authenticator":"This application is no longer maintained and has been replaced with com.belmoussaoui.Authenticator. Please create a backup and migrate to the new version",
    "com.github.bitseater.weather":"This application has been renamed to com.gitlab.bitseater.meteo",
    "com.github.bleakgrey.tootle":"The upstream project has been archived and is no longer getting updates.",
    "com.github.danrabbit.harvey":"This flatpak is unmaintained and out of date. It is recommended to install from a source officially supported by elementary.",
    "com.github.elfenware.badger":"Unfortunately, Badger will no longer receive updates on Flathub, but you can still download a Flatpak from https://appcenter.elementary.io/com.github.elfenware.badger. It will receive regular updates.\n\nTechnical details: This decision was made because we depend on the elementary SDK, which is not fully compatible with the elementary BaseApp available on Flathub, and may diverge further, causing maintenance issues. Thanks for using Badger, and apologies for the inconvenience.",
    "com.github.eudaldgr.elements":"Elements is no longer available as a Flatpak package.",
    "com.github.lainsce.coin":"This application is no longer maintained because the developer is no longer updating it.",
    "com.github.lainsce.timetable":"This application is no longer maintained because the developer is no longer updating it.",
    "com.github.lainsce.yishu":"This application is no longer maintained because the developer is no longer updating it.",
    "com.gitlab.KRypt0n_.an-anime-game-launcher":"We decided to move An Anime Game Launcher's flatpak to our own remote in order to avoid potential issues. Please see the project's GitLab page for instructions on how to switch.",
    "com.jvieira.tpt.Metronome":"The application has been deprecated as requested by its author.",
    "com.krillbite.ThePlan":"This application is no longer maintained because it has been superseded by the Steam version.",
    "com.vinszent.GnomeTwitch":"This application is no longer maintained because Twitch.tv updated their API and the application doesn't support it.",
    "cx.ring.Ring":"This application has been renamed to net.jami.Jami",
    "cz.pervoj.valdo-gtk":"Unfortunately, Valdo GTK will no longer receive updates on Flathub. For more information visit this page: https://github.com/pervoj/valdo-gtk#readme Thanks for using Valdo GTK!",
    "de.hoppfoundation.klassenzimmer":"This application is no longer maintained because the project ended on 30.10.2022",
    "io.bit3.ThreemaQT":"This application is no longer maintained, please use the official Desktop Client https://threema.ch/en/blog/posts/threema-for-desktop",
    "io.bit3.WhatsAppQT":"This application is no longer maintained, there are some good alternatives: https://flathub.org/apps/search/whatsapp",
    "io.elementary.Loki.BaseApp":"The BaseApp reached it's end-of-life, please migrate to io.elementary.BaseApp",
    "io.elementary.code":"This flatpak is unmaintained and out of date. It is recommended to install from a source officially supported by elementary.",
    "io.github.Fontforge":"This application has moved to a new appid. The new appid is org.fontforge.FontForge",
    "io.github.GnomeMpv":"This project has been renamed to io.github.celluloid_player.Celluloid",
    "io.github.harmonoid.harmonoid":"The application has been removed from Flathub.",
    "io.github.jkozera.ZevDocs":"ZevDocs is end-of-life and will no longer be receiving updates.",
    "io.github.lukajankovic.ambience":"This application is no longer maintained due to lack of time and motivation.",
    "io.github.mujx.Nheko":"This application is no longer maintained.",
    "io.github.prateekmedia.flutube":"Use PsTube instead.",
    "io.github.torbuntu.Leikr":"Temporal changes demand a shift in focus to other endeavors.",
    "nz.scuttlebutt.Patchwork":"The application is no longer maintained on Flathub.",
    "org.baedert.corebird":"Vital Twitter API endpoints have been retired and as a consequence, Corebird is unmaintained and will not work in the future.",
    "org.ethereum.Mist":"The Mist application has been deprecated",
    "org.eu.encom.matrique":"This application has been renamed to org.eu.encom.spectral, please uninstall this and install that",
    "org.glimpse_editor.Glimpse":"Glimpse is no longer maintained. See https://glimpse-editor.org/posts/a-project-on-hiatus/ for details.",
    "org.gnome.Books":"This application is no longer maintained because it is unmaintained upstream.",
    "org.gnome.Documents":"Documents is end-of-life and will no longer be receiving updates.",
    "org.gnome.FeedReader":"No longer maintained! Check out FeedReaders spiritual successor called: NewsFlash.",
    "org.gnome.Nautilus":"Nautilus is improving Flatpak integration before delivering it as stable, in the meantime the maintainers encourage you to test the development version from https://sdk.gnome.org/gnome-apps-nightly.flatpakrepo",
    "org.gnome.bijiben":"This application was replaced by org.gnome.Notes",
    "org.gnome.chess":"This application has been renamed to org.gnome.Chess",
    "org.gnome.ghex":"Application will been renamed to org.gnome.GHex",
    "org.kde.kate":"This application is no longer maintained since all the features are not properly supported in the flatpak version.",
    "org.racket_lang.DrRacket":"This extension will not be updated. Please use 'org.racket_lang.Racket' instead",
    "org.tabos.Twofun":"Application will been renamed to org.tabos.twofun",
    "sh.ppy.osulazer":"This application is no longer maintained."
}
```

> just not really sure if there's any point in showing EOL message itself

As you can see above, some just say that the App is no longer maintained. Others say, that the App has only been removed on Flathub but the App itself is still maintained. Some will tell you a other App that you can use. There are also some Apps which tell you the reason why they are no longer maintained, which might be interesting for the User. Overall I see no reason why we shouldn't show this Message, We have more than enough space.

>  let me see if I can dump that from the summary somehow

What did you mean?

>  the message should make it clear it's a message provided by developer

If you have another Idea how to structure this, please tell me.

> if your comfortable to join our matrix room, it might be easier to sort this out

I'm already in the Matrix Room, but I think GitHub is way better. There are different topics discussed in the Room, while here in the Comments we only talk about this specific PR. It's easier to follow the Discussion, especially if someone in the Future wants to know why this is implemented the Way it is. The Discussion n GitHub is also public and can be read by everyone while Matrix needs a Account to read.

**Comment by JakobDev** [comment]
@razzeee We are now post launch. Any updates?

**Comment by razzeee** [comment]
I would be fine with this, but I think having a designer take a look might be nice, even if it's not many elements.

I think not centering stuff might already help just top + center and maybe a box/card or warning thing?
Maybe one of these https://tailwindui.com/components/application-ui/feedback/alerts (I can code that, if we can agree on one)

File: frontend/src/fetchers.ts [hunk 1]
```diff
@@ -91,6 +92,25 @@ export async function fetchEolRebase(
...
+    const entryData = await fetch(`${EOL_MESSAGE_URL(appId)}`)
+    if (entryData.status === 200) {
+      entryJson = await entryData.json()
+    }
+  } catch (error) {
+    console.log(error)
+  }
+
+  if (!entryJson) {
+    console.log(`No eol message data`)
```
Comment (razzeee) [inline frontend/src/fetchers.ts:109]: This might be a bit too spammy

**Comment by razzeee** [comment]
image

image

File: frontend/src/components/application/EolMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const EolMessage: FunctionComponent<Props> = ({ message }) => {
+  const { t } = useTranslation()
+
+  return (
+    <>
+      <NextSeo noindex={true} />
+      <div className="mt-8 flex max-w-full flex-col px-[5%] md:px-[20%] 2xl:px-[30%]">
+        <Alert
+          type="warning"
+          icon={HiExclamationTriangle}
```
Comment (razzeee) [inline frontend/src/components/application/EolMessage.tsx:21]: Ideally, this would be the small Hi icon, but I don't think react-icons supports that yet.

**Pull request merged**